### PR TITLE
Adjust handling of zero stdev in loss functions

### LINKED ISSUE #3041: Sortino Hyperopt not working
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: Ubuntu 18.04
  * Python Version: Python 3.7.4
  * CCXT version: ccxt==1.23.30
  * Branch: Develop
  * Last Commit ID: 77944175e27bcd7851a83863f5df15f3577d61de

  
## Step 3: Describe the problem:

Running hyperopt with the new Sortino method is not working, exactly same Hyperopt file does work with Sharpe method

### Steps to reproduce:

  1. _____
  2. _____
  3. _____
  
### Observed Results:

see log below

### Relevant code exceptions or logs:

  ```
(.env) (base) andreas@andreas-G5-5590:~/freqtrade$ freqtrade hyperopt -c config_lever_bull.json --strategy Bblow_AP --hyperopt BblowHyperopt -e 1000 --spaces all --hyperopt-loss SortinoHyperOptLoss --timerange=20200212-20200307  --ticker-interval 15m
2020-03-09 13:14:57,174 - freqtrade.loggers - INFO - Verbosity set to 0
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Parameter -i/--ticker-interval detected ... Using ticker_interval: 15m ...
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 3 ...
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Parameter --timerange detected: 20200212-20200307 ...
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Using user-data directory: /home/andreas/freqtrade/user_data ...
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Using data directory: /home/andreas/freqtrade/user_data/data/binance ...
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Overriding ticker interval with Command line argument
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Using Hyperopt class name: BblowHyperopt
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Parameter --epochs detected ... Will run Hyperopt with for 1000 epochs ...
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Parameter -s/--spaces detected: ['all']
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Parameter -j/--job-workers detected: -1
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Parameter --min-trades detected: 1
2020-03-09 13:14:57,174 - freqtrade.configuration.configuration - INFO - Using Hyperopt loss class name: SortinoHyperOptLoss
2020-03-09 13:14:57,174 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2020-03-09 13:14:57,174 - freqtrade.configuration.check_exchange - INFO - Exchange "binance" is officially supported by the Freqtrade development team.
2020-03-09 13:14:57,175 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2020-03-09 13:14:57,175 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2020-03-09 13:14:57,176 - freqtrade.commands.optimize_commands - INFO - Starting freqtrade in Hyperopt mode
2020-03-09 13:14:57,176 - filelock - INFO - Lock 140312005427600 acquired on /home/andreas/freqtrade/user_data/hyperopt.lock
2020-03-09 13:14:57,176 - freqtrade.exchange.exchange - INFO - Instance is running with dry_run enabled
2020-03-09 13:14:57,176 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True}
2020-03-09 13:14:57,178 - freqtrade.exchange.exchange - INFO - Applying additional ccxt config: {'enableRateLimit': True, 'rateLimit': 500}
2020-03-09 13:14:57,180 - freqtrade.exchange.exchange - INFO - Using Exchange "Binance"
2020-03-09 13:14:57,928 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Binance'...
2020-03-09 13:14:57,947 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy Bblow_AP from '/home/andreas/freqtrade/user_data/strategies/Bblow_AP.py'...
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'ticker_interval' with value in config file: 15m.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'order_types' with value in config file: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USDT.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: 500.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'buy': 10, 'sell': 30}.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'use_sell_signal' with value in config file: True.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'sell_profit_only' with value in config file: False.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'ignore_roi_if_buy_signal' with value in config file: False.
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'400': 0.0, '200': 0.02, '80': 0.05, '40': 0.06, '0': 0.07}
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ticker_interval: 15m
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.1
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: False
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'buy': 'limit', 'sell': 'limit', 'stoploss': 'market', 'stoploss_on_exchange': False, 'stoploss_on_exchange_interval': 60}
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'buy': 'gtc', 'sell': 'gtc'}
2020-03-09 13:14:57,947 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USDT
2020-03-09 13:14:57,948 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: 500
2020-03-09 13:14:57,948 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 200
2020-03-09 13:14:57,948 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'buy': 10, 'sell': 30}
2020-03-09 13:14:57,948 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_sell_signal: True
2020-03-09 13:14:57,948 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_only: False
2020-03-09 13:14:57,948 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_buy_signal: False
2020-03-09 13:14:57,948 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2020-03-09 13:14:57,950 - freqtrade.resolvers.iresolver - WARNING - Could not import /home/andreas/freqtrade/user_data/hyperopts/BblowHyperopt1.py due to 'keyword can't be an expression (BblowHyperopt1.py, line 87)'
2020-03-09 13:14:57,951 - freqtrade.resolvers.iresolver - INFO - Using resolved hyperopt BblowHyperopt from '/home/andreas/freqtrade/user_data/hyperopts/BblowHyperopt.py'...
2020-03-09 13:14:57,953 - freqtrade.resolvers.iresolver - WARNING - Could not import /home/andreas/freqtrade/user_data/hyperopts/BblowHyperopt1.py due to 'keyword can't be an expression (BblowHyperopt1.py, line 87)'
2020-03-09 13:14:57,956 - freqtrade.resolvers.iresolver - INFO - Using resolved hyperoptloss SortinoHyperOptLoss from '/home/andreas/freqtrade/freqtrade/optimize/hyperopt_loss_sortino.py'...
2020-03-09 13:14:57,956 - freqtrade.optimize.hyperopt - INFO - Removing `/home/andreas/freqtrade/user_data/hyperopt_results/hyperopt_tickerdata.pkl`.
2020-03-09 13:14:57,956 - freqtrade.optimize.hyperopt - INFO - Removing `/home/andreas/freqtrade/user_data/hyperopt_results/hyperopt_results.pickle`.
2020-03-09 13:14:57,957 - freqtrade.optimize.hyperopt - INFO - Using optimizer random state: 33741
2020-03-09 13:14:57,959 - freqtrade.data.history.history_utils - INFO - Using indicator startup period: 200 ...
2020-03-09 13:14:57,984 - freqtrade.data.converter - INFO - Missing data fillup for BULL/USDT: before: 2474 - after: 2505
2020-03-09 13:14:58,004 - freqtrade.data.converter - INFO - Missing data fillup for ETHBULL/USDT: before: 2474 - after: 2505
2020-03-09 13:14:58,011 - freqtrade.data.history.idatahandler - WARNING - Missing data at start for pair EOSBULL/USDT, data starts at 2020-02-12 08:00:00
2020-03-09 13:14:58,024 - freqtrade.data.converter - INFO - Missing data fillup for EOSBULL/USDT: before: 2242 - after: 2273
2020-03-09 13:14:58,031 - freqtrade.data.history.idatahandler - WARNING - Missing data at start for pair XRPBULL/USDT, data starts at 2020-02-12 08:00:00
2020-03-09 13:14:58,043 - freqtrade.data.converter - INFO - Missing data fillup for XRPBULL/USDT: before: 2242 - after: 2273
2020-03-09 13:14:58,044 - freqtrade.optimize.backtesting - INFO - Loading data from 2020-02-09T22:00:00+00:00 up to 2020-03-07T00:00:00+00:00 (26 days)..
2020-03-09 13:14:58,079 - freqtrade.optimize.hyperopt - INFO - Hyperopting with data from 2020-02-09T22:00:00+00:00 up to 2020-03-07T00:00:00+00:00 (26 days)..
2020-03-09 13:14:58,086 - freqtrade.optimize.hyperopt - INFO - Found 12 CPU cores. Let's make them scream!
2020-03-09 13:14:58,086 - freqtrade.optimize.hyperopt - INFO - Number of parallel jobs set as: -1
2020-03-09 13:14:58,100 - freqtrade.optimize.hyperopt - INFO - Effective number of parallel workers used: 12
.

+--------+-----------+----------+--------------+-------------------------------+----------------+-------------+
|   Best |     Epoch |   Trades |   Avg profit |                        Profit |   Avg duration |   Objective |
|--------+-----------+----------+--------------+-------------------------------+----------------+-------------|
|   Best |    1/1000 |       31 |        0.54% |   83.73168355 USDT   (16.73%) |        119.0 m |    -2.63112 |
.

|   Best |    2/1000 |       29 |        1.54% |  223.13335677 USDT   (44.58%) |        165.5 m |    -21.1971 |
....

|   Best |    6/1000 |       12 |        1.82% |  109.45889211 USDT   (21.87%) |        117.5 m |        -inf |
..................2020-03-09 13:15:00,954 - freqtrade - ERROR - Fatal exception!
Traceback (most recent call last):
  File "/home/andreas/freqtrade/freqtrade/main.py", line 36, in main
    return_code = args['func'](args)
  File "/home/andreas/freqtrade/freqtrade/commands/optimize_commands.py", line 76, in start_hyperopt
    hyperopt.start()
  File "/home/andreas/freqtrade/freqtrade/optimize/hyperopt.py", line 613, in start
    asked = self.opt.ask(n_points=current_jobs)
  File "/home/andreas/freqtrade/.env/lib/python3.7/site-packages/skopt/optimizer/optimizer.py", line 382, in ask
    opt._tell(x, y_lie)
  File "/home/andreas/freqtrade/.env/lib/python3.7/site-packages/skopt/optimizer/optimizer.py", line 496, in _tell
    est.fit(self.space.transform(self.Xi), self.yi)
  File "/home/andreas/freqtrade/.env/lib/python3.7/site-packages/sklearn/ensemble/_forest.py", line 296, in fit
    y = check_array(y, accept_sparse='csc', ensure_2d=False, dtype=None)
  File "/home/andreas/freqtrade/.env/lib/python3.7/site-packages/sklearn/utils/validation.py", line 578, in check_array
    allow_nan=force_all_finite == 'allow-nan')
  File "/home/andreas/freqtrade/.env/lib/python3.7/site-packages/sklearn/utils/validation.py", line 60, in _assert_all_finite
    msg_dtype if msg_dtype is not None else X.dtype)
ValueError: Input contains NaN, infinity or a value too large
  ```

### PR DESCRIPTION
**Pull request by hroff-1902**
Fixes #3041, in the quick&dirty way.

I'll refactor handling of "bad" values returned from the loss function method later.

### CODE REVIEW

File: freqtrade/optimize/hyperopt_loss_sortino.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def hyperopt_loss_function(results: DataFrame, trade_count: int,
         results.loc[total_profit < 0, 'downside_returns'] = results['profit_percent']
         down_stdev = np.std(results['downside_returns'])
 
-        if np.std(total_profit) != 0.0:
+        if down_stdev != 0:
```
Comment (xmatthias) [inline freqtrade/optimize/hyperopt_loss_sortino.py]: are you sure about changing `0.0` to `0`?

Comment (hroff-1902) [reply freqtrade/optimize/hyperopt_loss_sortino.py]: should not be a problem, it's a constant and will be cast to python if needed comparing float with it

Comment (hroff-1902) [reply freqtrade/optimize/hyperopt_loss_sortino.py]: this should call 
```
(down_stdev).__neq__(0)
```
i.e. call np.float64.__neq__, which knows how to handle ints...

Comment (hroff-1902) [reply freqtrade/optimize/hyperopt_loss_sortino.py]: if I'm not wrong here, though...

**Pull request merged**